### PR TITLE
feat(ui): Add `startedAt` and `finishedAt` variables to configurable links. Fixes #3898

### LINKED ISSUE #3898: Improve Links feature
# Summary

- #2443
- Support `${status.startedAt}` and `${status.finishedAt}` as substitutable variables in `links`.
  - Links just support `${metadata.name}`, `${metadata.namespace}` variables now.

# Use Cases

- For example, administrators can provide a link to Log Dashboard within the time window.

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by posquit0**
### Background

- #3898 

### Checklist

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

**Comment by posquit0** [comment]
@alexec Could you please to give me the advice to fix the broken CI?

File: ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx [hunk 1]
```diff
@@ -247,6 +247,10 @@ export class ArchivedWorkflowDetails extends BasePage<RouteComponentProps<any>,
     }
 
     private openLink(link: Link) {
-        document.location.href = link.url.replace('${metadata.namespace}', this.state.workflow.metadata.namespace).replace('${metadata.name}', this.state.workflow.metadata.name);
```
Comment (alexec) [inline ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx:250]: Looks good! Anyway we can refactor this into a new React component, e.g.` <LinkButton workflow={this.state.workflow}/>` to reduce duplication.

Comment (alexec) [reply ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx:250]: or maybe better still:

```
<Links links={links} scope='workflow' object={this.status.workflow}/>
```

and

```
<Links links={links} scope='pod' object={node}/>
```

Comment (posquit0) [reply ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx:250]: Hmmm... I'm not familiar with React. Could you support me to refactor that?

Comment (posquit0) [reply ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx:250]: @alexec Buttons for `links` is rendered via `actionMenu` property of `Page` component which is from [argo-ui](https://github.com/argoproj/argo-ui). And `Page` component actually render buttons for `links` via [`Topbar` component](https://github.com/argoproj/argo-ui/blob/master/src/components/top-bar/top-bar.tsx#L98).

How can I refactor this code without modifying `react-ui`? It doesn't look easy. 😭

Comment (alexec) [reply ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx:250]: ah - forget about it then

**Review by alexec — CHANGES_REQUESTED** [review]
Can you `yarn lint` and we should be good.

**Comment by alexec** [comment]
```
 FAIL: TestNextRuntime
```

This test is fails about 1/50 builds - it is flaky - @sarabala1979 @simster7

File: ui/src/app/workflows/components/workflow-node-info/workflow-node-info.tsx [hunk 2]
```diff
@@ -120,7 +120,11 @@ export const WorkflowNodeSummary = (props: Props) => {
                         .map(link => (
                             <a
                                 className='argo-button argo-button--base-o'
-                                href={link.url.replace('${metadata.namespace}', props.workflow.metadata.namespace).replace('${metadata.name}', props.node.id)}>
+                                href={link.url
+                                    .replace('${metadata.namespace}', props.workflow.metadata.namespace)
+                                    .replace('${metadata.name}', props.node.id)
+                                    .replace('${status.startedAt}', props.workflow.status.startedAt)
```
Comment (alexec) [inline ui/src/app/workflows/components/workflow-node-info/workflow-node-info.tsx:126]: should this be the started finished at dates of the node not the workflow?

**Comment by simster7** [comment]
>  This test is fails about 1/50 builds - it is flaky - @sarabala1979 @simster7

Should be exactly 1/60 builds :) given that it deals with seconds in a minute. Fixed with #3979

File: ui/src/app/workflows/components/workflow-node-info/workflow-node-info.tsx (hunk 2 above)
Comment (posquit0) [reply ui/src/app/workflows/components/workflow-node-info/workflow-node-info.tsx]: thx! I fixed!

**Review by alexec — APPROVED** [review]
🚀

**Comment by alexec** [comment]
Merged!

**Pull request merged**